framework/src/media/MediaRecorderImpl.h

```diff
@@ -68,28 +68,38 @@ class MediaRecorderImpl : public enable_shared_from_this<MediaRecorderImpl>
 
 	recorder_result_t create();
 	recorder_result_t destroy();
-
 	recorder_result_t prepare();
 	recorder_result_t unprepare();
+
 	recorder_result_t start();
 	recorder_result_t pause();
 	recorder_result_t stop();
 
 	int getVolume();
 	recorder_result_t setVolume(int vol);
-
 	recorder_result_t setDataSource(std::unique_ptr<stream::OutputDataSource> dataSource);
 	recorder_state_t getState();
-	void setState(recorder_state_t state);
-	void setObserver(std::shared_ptr<MediaRecorderObserverInterface> observer);
-	
+	recorder_result_t setObserver(std::shared_ptr<MediaRecorderObserverInterface> observer);
 	void notifySync();
 	void notifyObserver(observer_command_t cmd);
 	void capture();
 
+private:
+	void createRecorder(recorder_result_t& ret);
+	void destroyRecorder(recorder_result_t& ret);
+	void prepareRecorder(recorder_result_t& ret);
+	void unprepareRecorder(recorder_result_t& ret);
+	void startRecorder();
+	void pauseRecorder();
+	void stopRecorder(bool);
+	void getVolumeRecorder(int& ret);
```

Comment:
getRecorderVolume / setRecorderObserver / setRecorderDataSource is better naming.